### DISCUSSION TITLE
How to make really quite output for console application?

### DISCUSSION THREAD

**Discussion by max-petrovich** [symfony/symfony#51965] [Q&A]
The `-q` option still displays errors. I found the NullOutput object but can't figure out how to use it.

Q: How to use `NullOutput` in my console application?
WHY: Symfony still displays the error in the logs; I don’t need to additionally see both the logs and the formatted output of the application itself

**Comment by ghost** [comment]
Please provide a Symfony version you use and information what kind of errors are displayed exactly.

**Comment by max-petrovich** [comment]
All versions (i think). 

Symfony returns unformatted (e.g. json) error output (if one occurs). And also logs the same error. Then, if the logs are also sent to stdout, when calling the console command (in which an error occurs), we see the logs in stdout (in json, for example) and unformatted error output. And the -q mode does not affect this in any way; the error is still output to stdout. And I ONLY need logs in output.

**Comment by yblatti** [comment]
In my experience, it's quite common that a "quiet command" is only quiet when "everything is fine", but still output errors (in stderr stream).

I would use output redirection on call, something like `bin/console app:shoule-quiet -q 2>stderr.log 1>stdout.log`.
Would that be possible in your case ?

**Reply by ghost** [reply]
If that's true then it should be mentioned in the `--quiet` option description, because it's a big difference.

**Reply by max-petrovich** [reply]
My logs are formatted in json and then collected by the collector. And if there is an error, symfony displays an unformatted log to the console. But it also writes an error in the logs. It would be more logical to not display errors at all in --quite mode
